### ISSUE TITLE
Variations containing a single quote broken after WC 2.4 upgrade

### ISSUE THREAD

**Issue by douglsmith** [woothemes/woocommerce#9040]
After upgrading to WC 2.4, I'm running into problems with variations that are based on attributes having a single quote (used as an apostrophe) in the name.

Old products that had a single quote in the variation name can be added to the cart but the attribute containing the quote will be left off. The variation display is also broken when editing the product. Here are a few screen shots.

<img width="946" alt="pastedgraphic-18" src="https://cloud.githubusercontent.com/assets/2446310/9665537/9a84847e-5236-11e5-8986-6158f7cf871d.png">

<img width="761" alt="pastedgraphic-17" src="https://cloud.githubusercontent.com/assets/2446310/9663964/c293b50c-522c-11e5-9604-5e917d81c695.png">

And here's what happens on the front end.

<img width="492" alt="pastedgraphic-15" src="https://cloud.githubusercontent.com/assets/2446310/9664012/f490b776-522c-11e5-8f02-da7df383b84a.png">

<img width="324" alt="pastedgraphic-16" src="https://cloud.githubusercontent.com/assets/2446310/9664058/386b6ba8-522d-11e5-8198-0f2625cd78b9.png">

You can add a new attribute with an apostrophe and assign it to a variation. It appears to be saved but when the product or draft is saved then the attribute is completely gone. Here are some screen shots showing what happens.

<img width="869" alt="pastedgraphic-20" src="https://cloud.githubusercontent.com/assets/2446310/9671677/3b172626-525a-11e5-8cd5-f17b477b57d9.png">

<img width="943" alt="pastedgraphic-22" src="https://cloud.githubusercontent.com/assets/2446310/9670949/f2fdc0b6-5254-11e5-945b-5ea054cd5703.png">

<img width="904" alt="pastedgraphic-21" src="https://cloud.githubusercontent.com/assets/2446310/9671695/6d648c40-525a-11e5-95c9-22bdcfa87c8d.png">

And here's what happens on the front end.

<img width="741" alt="pastedgraphic-23" src="https://cloud.githubusercontent.com/assets/2446310/9671365/035bb8f2-5258-11e5-8b43-10c559076112.png">

**Comment by mikejolley** [comment]
Here is what I just tried to no avail.

1. Installed 2.3.12
2. Created new variable product
3. Added text attribute named Test with value "Doug&#39;d | Test"
4. Added 2 variations; one for each

Then I switched to 2.4 and compared the results. Variations stayed assigned. Attributes remained the same. Frontend worked.

Then I saved the attributes and repeated the test - same result. 

It was only when I changed a variation and saved, did the attribute with ' get unassigned. I didn't see the slashes you showed.

**Comment by mikejolley** [comment]
Can you see if that commit clears up admin variations panel? i.e. can you set the attribute and save.

**Comment by douglsmith** [comment]
I had removed all of the apostrophes and reworked my attributes during troubleshooting and getting screen shots. That was on both my test and production sites. I went back to make them fail again before applying your above commit and could not make them do so. This had not been a factor before but I had never removed the quotes from every product. I am now able to add apostrophes back in and not have a problem. So it looks like I had some old corrupt data that was a factor in this.

I was eventually able to reproduce one of the problems, though. If the apostrophe is not a single quote character but the &#039; entity, it still causes variations to disappear. I never originally entered that entity and when I now save, my single quote is not replaced by the entity. I'm guessing that at some time WordPress or WooCommerce was forcing that entity conversion.

After applying your above commit, I can confirm that it does fix any problem I've been able to currently reproduce using the &#039; entity.
